packages/grid/src/GridRenderer.js

```diff
@@ -1142,16 +1158,67 @@ class GridRenderer {
     context.fillStyle = headerBackgroundColor;
     context.fillRect(0, 0, width, columnHeaderHeight);
 
-    // Draw the separators
+    context.fillStyle = headerColor;
+
+    // Visible columns.
+    for (let i = 0; i < visibleColumns.length; i += 1) {
+      const column = visibleColumns[i];
+      const columnWidth = visibleColumnWidths.get(column);
+      const x = visibleColumnXs.get(column) + gridX;
+      this.drawColumnHeader(context, state, column, x, columnWidth);
+    }
+
+    if (containsFrozenColumns) {
+      floatingLeftColumnsWidth =
+        visibleColumnXs.get(floatingLeftColumnCount - 1) +
+        visibleColumnWidths.get(floatingLeftColumnCount - 1);
+
+      // Frozen columns' background
+      context.fillStyle = headerBackgroundColor;
+      context.fillRect(0, 0, floatingLeftColumnsWidth, columnHeaderHeight);
+
+      // Frozen columns.
+      context.fillStyle = headerColor;
+      for (let i = 0; i < floatingColumns.length; i += 1) {
+        const column = floatingColumns[i];
+        const columnWidth = visibleColumnWidths.get(column);
+        const x = visibleColumnXs.get(column) + gridX;
+        this.drawColumnHeader(context, state, column, x, columnWidth);
+      }
+    }
+
+    // Draw the separators, visible columns then floating columns.
     if (headerSeparatorColor) {
       context.strokeStyle = headerSeparatorColor;
       context.beginPath();
       const hiddenColumns = [];
+
+      // Draw visible column separators.
       let isPreviousColumnHidden = false;
       for (let i = 0; i < visibleColumns.length; i += 1) {
         const column = visibleColumns[i];
         const columnX = visibleColumnXs.get(column);
         const columnWidth = visibleColumnWidths.get(column);
+
+        if (!(columnX < floatingLeftColumnsWidth - columnWidth)) {
```

Comment:
Actually after inspecting further, there is some overlap occurring when I'm testing in the styleguide:

![image](https://user-images.githubusercontent.com/4505624/141174561-aef5b2a8-1a21-4aee-a812-9c5af31ca0a7.png)

Reply:
I really can't repro this. We might need to screen-share...I'll post the rest of the review fixes in the meantime.

Reply:
Looks like it's only reproducible when the row headers are visible, which we have in the style guide. I set a breakpoint in the drawColumnHeaders when it was filling in the background to figure out the floating background it was drawing was a bit off:

![image](https://user-images.githubusercontent.com/4505624/141233252-3b9a5b5a-7bd6-4f4f-8189-829f11bc7aac.png)

Just needed to offset the background drawing by `gridX` instead of `0`